DatadogSessionReplay/Tests/DatadogSessionReplayTests/DatadogCoreIntegration/RequestBuilder/RequestBuilderTests.swift

```diff
@@ -5,11 +5,14 @@
  */
 
 import XCTest
+import Datadog
 @testable import DatadogSessionReplay
+@testable import TestUtilities
 
 // swiftlint:disable empty_xctest_method
 class RequestBuilderTests: XCTestCase {
     func testWhenCustomUploadURLIsNotSet_itCreatesRequestsToAppropriateDatadogSite() {
+        _ = DatadogContext.mockAny()
         // TODO: RUMM-2690
         // Implementing this test requires creating mocks for `DatadogContext` (passed in `FeatureRequestBuilder`),
         // which is yet not possible as we lack separate, shared module to facilitate tests.
```

Comment:
nit/ We could try adding at least one actual test in SR to make sure it works end-to-end. `RequestBuilder` might not be the easiest one, but I made following try with `SR/../WriterTests` and it works perfectly well ⭐:

```diff
     func testWhenFeatureScopeIsConnected_itWritesRecordsToCore() {
-        // TODO: RUMM-2690
-        // Implementing this test requires creating mocks for `DatadogContext` (passed in `FeatureScope`),
-        // which is yet not possible as we lack separate, shared module to facilitate tests.
+        // Given
+        let core = PassthroughCoreMock()
+        let writer = Writer()
+
+        // When
+        writer.startWriting(to: core)
+
+        // Then
+        writer.write(nextRecord: EnrichedRecord(rumContext: .mockRandom(), records: .mockRandom()))
+        writer.write(nextRecord: EnrichedRecord(rumContext: .mockRandom(), records: .mockRandom()))
+        writer.write(nextRecord: EnrichedRecord(rumContext: .mockRandom(), records: .mockRandom()))
+
+        XCTAssertEqual(core.events(ofType: EnrichedRecord.self).count, 3)
     }
```